tests/JsonFileHelperTest.php

```diff
@@ -82,4 +82,19 @@ public function testWriteToFilePrettyPrintedSuccessfully(): void
         );
         unlink($filePath);
     }
+
+    public function testWriteToNonExistingDirectoryThrowsException(): void
```

Comment:
Tohle podle mého je nesprávné chování - právě, že by to mělo šetřit čas a tu složku vygenerovat. Co, @odinuv ?

Reply:
@odinuv Psal, ze ta slozka `out`, tam vzdy bude a kdyz ne, tak se to ma probublat do ApplicationErroru.

Reply:
zalezi za co chcete ten jsonfilehelper mit, jestli ma slouzit jen pro potreby php-component, tak to slozku vytvaret nemusi
jestli ho chcete pouzivat i jinde, tak je to na zvazeni

Reply:
Není to outfile writer. Je to generický dumper json filů. Takže s tím stejně tak dobře můžu chtít zapsat třeba /out/tables/mytable/some.manifest

Reply:
nemohli bysme tady v tomhle PR nechat symfony encoder a ten helper resit nekde zvlast? 
protoze kdyz tohle mergnem, tak bude state pouzivat helper, a manifset symfony encoder, anebo to teda zmenit i tam, ale to u zase dela ten PR 3 veci

Reply:
jestli je genericky, tak by se mel pouzivat i pro ty manifesty a porad bych se primlouval za to aby mel staticky metody

Reply:
OK, statické metody souhlas. A ten JsonHelper můžem dořešit jinde.

Reply:
(vyřešit) = tady být teda nemusí

Reply:
Z tohoto uz vznika co me napadalo udelat hned na zacatku. Mini repo, kde bude jen tato trida se statickyma metodama a vubec to nemusi byt soucasti php-component. S tim, ze tam budou i vyhazovat pripadne exceptiony. A muzeme to pouzit uplne kdekoli a nebudeme potrebovat php-component.

V tomto stavu ale jsem pro to, aby to ty slozky automaticky vytvorilo + predelat na staticke metody.

Reply:
Jo, to jsem si plánoval úplně ze začátku, ale přišlo mi to moc převratné :D Byly by to json-utils a rovnou by tam byla i nějaká triviální implementace encode/decode aby se nemusel tahat celý SF serializer...

Reply:
Uz mame https://github.com/keboola/php-utils, co to pridat tam do `Keboola\Utils\JsonFile\Reader` a `Keboola\Utils\JsonFile\Reader`?

Reply:
Já musím přiznat, že php-utils  nemám moc rád - je to hrozný pelmel. Je tam random úplně všechno. A vlastně nikdo neví, co přesně. A kupodivu už tam json decode je https://github.com/keboola/php-utils/blob/master/src/Keboola/Utils/jsonDecode.php :D

Reply:
Ale jo, dejme to tam...

Reply:
taky nemam php-utils rad a dal bych to radsi nekam jinam

Reply:
Toho jsem si take vsiml, proto bych to zadelal do slozky, at je to od toho pelmelu odstinene.

Ale lepsi nez se tomu php-utils vyhybat jak cert krizi protoze je v nem chaoz, bychom mu mohli timto dat impuls k oziveni. Nez se na to divat jak to hnije.

Reply:
oziveni php-utils vyzaduje dohledat kde vsude se ty funkce tam definovany opravdu pouzivaji a potom s tim nejak nalozit, tipuju, ze se to cely pouziva tak ve 3 appkach a kazda z toho pouziva uplne disjunktni mnozinu funkci :)